src/common/options/CMakeLists.txt

```diff
@@ -104,8 +104,10 @@ add_options(rgw)
 
 add_library(common-options-objs OBJECT
   ${common_options_srcs})
-add_custom_target(legacy-option-headers
-  DEPENDS ${legacy_options_headers})
+add_library(legacy-option-headers INTERFACE)
+target_sources(legacy-option-headers
```

Comment:
Hi Kefu,

This breaks configuring (i.e. `do_cmake.sh` -- it doesn't get to the actual build) for me:
```
CMake Error at src/common/options/CMakeLists.txt:108 (target_sources):
  target_sources may only set INTERFACE properties on INTERFACE targets
...
-- Configuring incomplete, errors occurred!
See also "/home/dis/src/ceph-b/build/CMakeFiles/CMakeOutput.log".
See also "/home/dis/src/ceph-b/build/CMakeFiles/CMakeError.log".
+ exit 1
```
It's an old Ubuntu Focal box:
```
$ cmake --version
cmake version 3.16.3
```

Reply:
As expected, reverting this PR, https://github.com/ceph/ceph/pull/57583 and https://github.com/ceph/ceph/pull/57693 fixes it.

Reply:
@tchaikov ping ^^

Reply:
sorry for the breakage and for missing your comment. created #58078